util/patricia_trie/src/lookup.rs

```diff
@@ -55,7 +63,8 @@ impl<'a, Q: Query> Lookup<'a, Q> {
 			// without incrementing the depth.
 			let mut node_data = &node_data[..];
 			loop {
-				match Node::decoded(node_data)? {
+				// TODO: see comment on `TrieError::DecoderError` for what to do here
+				match C::decode(node_data).map_err(|e| TrieError::DecoderError(hash, format!("{:?}",e)))? {
```

Comment:
if `TrieError::DecoderError` is generic over decoder error type as well, this makes things easy